components/projects/Admin.vue

```diff
@@ -0,0 +1,331 @@
+          this.project.docs.push({
+            name: 'Project Report',
+            url: this.project_report
+          })
+        }
+        if (this.poster.length > 0) {
+          this.project.docs.push({
+            name: 'Poster',
+            url: this.poster
+          })
+        }
+        if (this.presentation.length > 0) {
+          this.project.docs.push({
+            name: 'Presentation',
+            url: this.presentation
+          })
+        }
+        if (this.project.docs.length === 0) {
+          delete this.project.docs
+        }
+      }
+    },
+    onPickFile() {
+      this.$refs.fileInput.click()
+    },
+    onMultiFilePick() {
+      this.$refs.multiFileInput.click()
+    },
+    onFilePicked(event) {
+      const files = event.target.files
+      const filename = files[0].name
+      if (filename.lastIndexOf('.') <= 0) {
+        return alert('Please upload a valid file.')
+      }
+      const fileReader = new FileReader()
+      fileReader.addEventListener('load', () => {
+        this.imageUrl = fileReader.result
+      })
+      fileReader.readAsDataURL(files[0])
+      this.image = files[0]
+    },
+    async onMultiFilePicked(event) {
+      try {
+        if (this.project.name === '') {
+          alert('Please enter the name of the project first!')
+        } else {
+          this.uploadRemaining = true
+          const ID = this.getID(this.project.name)
+          const files = event.target.files
+          console.log(files)
+          for (let i = 0; i < files.length; i++) {
+            const filename = files[i].name
+            const response = await firebase
+              .storage()
+              .ref('projects/' + ID + '/' + filename)
+              .put(files[i])
+            const downloadURL = await response.ref.getDownloadURL()
+            this.project.images.push(downloadURL)
+            this.uploadRemaining = !(
+              files.length === this.project.images.length
+            )
+          }
+        }
+      } catch (err) {
+        console.log(err)
+      }
+    },
+    async saveProject() {
+      try {
+        if (this.nonValid()) {
+          alert('Entered details are invalid. Please try again.')
+          return
+        } else if (this.imageUrl.length <= 0) {
+          alert("Please upload your project's image.")
+          return
+        }
+        this.loading = true
+        if (this.project.youtube.length <= 0) {
+          delete this.project.youtube
+        }
+        const filename = this.image.name
+        const ext = filename.slice(filename.lastIndexOf('.'))
+        this.setDocs()
+        this.project.id = this.getID(this.project.name)
+        const response = await this.$firebase
+          .database()
+          .ref('projects')
+          .push(this.project)
+        const key = response.key
+        const storageResponse = await firebase
+          .storage()
+          .ref('projects/' + this.project.id + '/' + key + ext)
```

Comment:
See this line. Project 'Safina Abdoz' will be in projects/safina-abdoz

Reply:
I'm not talking about the code implementation. I'm talking about the security rules